clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -235,7 +235,6 @@ public void addAll(final List<UnsentRequest> requests) {
 
     public void add(final UnsentRequest r) {
         Objects.requireNonNull(r);
-        r.setTimer(this.time, this.requestTimeoutMs);
```

Comment:
We provide the `Timer` when creating the `UnsentRequest`, so we don't need to add it afterward.

Reply:
It _shouldn't_. My preference is to make data as unmutable as possible. Setting the `Timer` on creation vs. after the fact seems cleaner.

Reply:
uhm well there is a difference: before these changes, the timeout was defined at the `NetworkClientDelegate` level and not per request, which made sense I would say, since it's the same `REQUEST_TIMEOUT_MS_CONFIG` that needs to be applied to all UnsentRequests, so it was applied here on this ln 238 when any request was added. 

Now we're saying that each UnsentRequest may have a diff timeout, but I see we end up with each manager taking that same REQUEST_TIMEOUT_MS_CONFIG config and passing it to each request. 

Outcome is the same, but actually the original version of applying the REQUEST_TIMEOUT_MS_CONFIG to all requests in the network layer seemed simpler I would say (unless I'm missing a gain of this option of having the REQUEST_TIMEOUT_MS_CONFIG at the managers level).

Reply:
I see from the commits that we reverted back to the original way, where the request timeout is defined at the networkclient level common for all requests, and passed onto the request on add, nice, let's just update the PR description then (we can remove point 4)

Reply:
Thanks for the reminder to update the PR description, @lianetm! Done.